File: algorithms/299.py
# https://leetcode.com/problems/bulls-and-cows


class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        s, g = list(secret), list(guess)
        A, B = 0, 0

        i = 0
        while i < len(s):
            if g[i] == s[i]:
                A += 1
                del s[i]
                del g[i]
            else:
                i += 1
        i = 0
        while i < len(s):
            if g[i] in s:
                B += 1
                s.remove(g[i])
                del g[i]
            else:
                i += 1

        return "%dA%dB" % (A, B)

File: algorithms/299.swift
// https://leetcode.com/problems/bulls-and-cows

class Solution {
    func getHint(secret: String, _ guess: String) -> String {
        var A = 0
        var B = 0
        var counter = [Int](count: 10, repeatedValue:0)
        var s_ = Array(secret.characters)
        var g_ = Array(guess.characters)
        for i in 0..<s_.count {
            let ds = Int(String(s_[i]))!
            let dg = Int(String(g_[i]))!
            if ds == dg {
                A += 1
            } else {
                counter[ds] += 1
                counter[dg] -=  1
                if counter[ds] <= 0 { B += 1 }
                if counter[dg] >= 0 { B += 1 }
            }
        }
        return "\(A)A\(B)B"
    }
}
